### PR TITLE
improved/robust version for the iso2mesh option within process_fem_mesh

### PR DESCRIPTION
**Pull request by tmedani**
Hi all,

This PR is about the issue observed when the iso2mesh option is used, and surf2mesh produces more element labels than the input surfaces.
link to the reported issue: https://neuroimage.usc.edu/forums/t/duneuro-error-calculation/28965/6?u=tmedani

The change on this code will identify any output label based on the distance comparison between the inputs surfaces and the outputs labels.
I believe that is much robust than the previous version.

This change can be also useful for custom fem mesh generation for sEEG/iEEG models.

Thanks :)

### CODE REVIEW

File: toolbox/process/functions/process_fem_mesh.m [hunk 1]
```diff
@@ -455,6 +456,150 @@
...
+                        sSubject.Surface(sSubject.iScalp).FileName};
+                    TissueLabels = {'brain', 'skull', 'scalp'};
+                else
+                    errMsg = [errMsg, 'Method "' OPTIONS.Method '" requires three surfaces: head, inner skull and outer skull.' 10 ...
+                        'Create them with process "Generate BEM surfaces" first.'];
+                    return;
+                end
+            % If surfaces are given: get their labels and sort from inner to outer
+            else
+                % Get tissue label
```
Comment (ftadel) [inline toolbox/process/functions/process_fem_mesh.m:483]: Here you removed the sorting and relabelling of the labels based on their names.
In some of the tests I made, the tissues ended up being incorrectly labelled in the FEM file...

For example, with the labels from MNE: `outer_skin.surf`, `outer_skull.surf`, `inner_skull.surf`
In the output FEM file, the layers don't get renamed in `scalp`, `skull`, `csf`, and they get incorrectly colored in the display.

Before:
image

After:
image

**Review by ftadel — CHANGES_REQUESTED** [review]
Since this is probably still an algorithm that we'll need to refine, I made it a new option `iso2mesh-2021` instead of editing the existing code.

There are some new issues to fix with the labeling of the different layers.

**Comment by tmedani** [comment]
Hi Francois, 
Thanks for reviewing  this, 
I think having both options is better (iso2mesh and iso2mesh2021),
if the first fails, one can tests the second. 

Regarding the issue that you highlighted, I will check it.

**Comment by tmedani** [comment]
Hi Francois, 

by a bad command line, I removed the change of the of this PR, 
by any chance do you still have the changes that you proposed for iso2mesh2021 version locally?

**Comment by ftadel** [comment]
No, I deleted the branch on my computer.
It sucks, there were quite a lot of edits in there...

When you want to work on a particular feature to merge with the master Brainstorm branch, create a separate branch on your own git local repository, instead of working only from your master branch. One branch per feature.
This way, simultaneous work on different features won't interfere with the reviewing process of individual PRs.

Example: We're working with Raymundo on the support for a new file format, he created a new branch, pushed it, we reviewed and edited it together, opened a PR for it (https://github.com/brainstorm-tools/brainstorm3/pull/416), and now he can keep on working on other things in different branches while this goes through the reviewing process.

Good luck with rewriting everything...

**Comment by tmedani** [comment]
Thanks, Francois, I used to work with branches before but not at this time. 
I believe that I can reproduce the code.
I will push a new PR next week.

**Pull request closed without merging**